pintc/tests/storage/invalid_state_var_storage_type.pnt

```diff
@@ -0,0 +1,31 @@
+type a = { ( bool => bool ), int };
+
+storage {
+    b: { ( int => int ), int },
+    c: { a, int },
+}
+
+predicate test {
+    state d = storage::b;
+    state e = storage::c;
+}
+
+// parsed <<<
+// type ::a = {( bool => bool ), int};
+// storage {
+//     b: {( int => int ), int},
+//     c: {::a, int},
+// }
+//
+// predicate ::test {
+//     state ::d = storage::b;
+//     state ::e = storage::c;
+// }
+// >>>
+
+// typecheck_failure <<<
+// state variables cannot have storage types
+// @122..142: found state variable of storage type ( int => int ) here
```

Comment:
so the error is a little misleading.. The type of the stat variable is not `( int => int )`. It's the whole tuple. However, the tuple itself contains the storage type. So we should reword this. I know you reused the old error variant `StateVarHasStorageType` but I don't think it fits anymore.

Reply:
Just updated. Instead of changing the error, I changed it so that we print the `parent_ty` if any `children_ty` are found to be a `map` or `vector`.

Ex. 
```
Error: state variables cannot have storage types
   ╭─[test.pnt:9:5]
   │
 9 │     state d = storage::b;
   │     ──────────┬─────────  
   │               ╰─────────── found state variable of storage type {( int => int ), int} here
───╯
Error: state variables cannot have storage types
    ╭─[test.pnt:10:5]
    │
 10 │     state e = storage::c;
    │     ──────────┬─────────  
    │               ╰─────────── found state variable of storage type {::a ({( bool => bool ), int}), int} here
────╯
```

Let me know if that's more appropriate or if you have something else in mind.

Reply:
In terms of rewording instead of printing the `parent_ty`, what do you think of:
```
Error: state variables cannot have storage types
   ╭─[test.pnt:9:5]
   │
 9 │     state d = storage::b;
   │     ──────────┬─────────  
   │               ╰─────────── state variable cannot have nested type of ( int => int )
───╯
```

I can then differentiate between the errors based on if its the first pass or subsequent ones.

Reply:
No what you have is fine. However, maybe adding a note would be helpful? Something like:
```
Note:  type of state variable is `{( int => int ), int}` which depends on the storage type `( int => int )`.
```

Reply:
```
Error: state variables cannot have storage types
   ╭─[test.pnt:9:5]
   │
 9 │     state d = storage::b;
   │     ──────────┬─────────  
   │               ╰─────────── found state variable of storage type {( int => int ), int} here
   │ 
   │ Note: type of state variable is `{( int => int ), int}` which depends on the storage type `( int => int )`.
───╯
Error: state variables cannot have storage types
    ╭─[test.pnt:10:5]
    │
 10 │     state e = storage::c;
    │     ──────────┬─────────  
    │               ╰─────────── found state variable of storage type {::a ({( bool => bool ), int}), int} here
    │ 
    │ Note: type of state variable is `{::a ({( bool => bool ), int}), int}` which depends on the storage type `( bool => bool )`.
────╯
```
seems a bit redundant to me. What do you think of
```
Error: state variables cannot have storage types
   ╭─[test.pnt:9:5]
   │
 9 │     state d = storage::b;
   │     ──────────┬─────────  
   │               ╰─────────── found state variable of storage type {( int => int ), int} here
   │ 
   │ Note: type of state variable is `{( int => int ), int}` which depends on the storage type `( int => int )`.
───╯
Error: state variables cannot have storage types
    ╭─[test.pnt:10:5]
    │
 10 │     state e = storage::c;
    │     ──────────┬─────────  
    │               ╰─────────── found state variable of storage type {::a ({( bool => bool ), int}), int} here
    │ 
    │ Note: type of state variable depends on the storage type `( bool => bool )`.
────╯
```

Reply:
Yeah that's reasonable